src/base/Pie.hh

```diff
@@ -203,47 +214,11 @@ class Pie
     //@}
 };
 
-//---------------------------------------------------------------------------//
-} // namespace celeritas
-
-//---------------------------------------------------------------------------//
-//! \cond
-#define CELER_PIE_TYPE_(CLSNAME, OWNERSHIP, MEMSPACE) \
-    CLSNAME<::celeritas::Ownership::OWNERSHIP, ::celeritas::MemSpace::MEMSPACE>
-//!\endcond
-
-/*!
- * \def CELER_PIE_STRUCT
- *
- * Define an anonymous struct that holds sets of values and references on
- * host and device. This is meant for collections of Pies in a struct
- * that's templated only on the ownership and memory space.
- *
- * Example:
- * \code
- * class FooParams
- * {
- *  public:
- *   using PieDeviceRef = FooPies<Ownership::const_reference,
- *                                MemSpace::device>;
- *
- *   const PieDeviceRef& device_pointers() const {
- *    return pies_.device_ref;
- *   }
- *  private:
- *   CELER_PIE_STRUCT(FooPies, const_reference) pies_;
- * };
- * \endcode
- */
-#define CELER_PIE_STRUCT(CLSNAME, REFTYPE)                    \
-    struct                                                    \
-    {                                                         \
-        CELER_PIE_TYPE_(CLSNAME, value, host) host;           \
-        CELER_PIE_TYPE_(CLSNAME, value, device) device;       \
-        CELER_PIE_TYPE_(CLSNAME, REFTYPE, host) host_ref;     \
-        CELER_PIE_TYPE_(CLSNAME, REFTYPE, device) device_ref; \
-    }
+//! Pie for data of type T but indexed by ThreadId
+template<class T, Ownership W, MemSpace M>
+using StatePie = Pie<T, W, M, ThreadId>;
```

Comment:
Maybe:
```suggestion
using LocalPie = Pie<T, W, M, ThreadId>;
```
?

Reply:
I don't think that nomenclature makes sense -- the data is still in global memory. The difference is that it's being indexed by ThreadId rather than `OpaqueId<T>`, so whereas `FooParams` define `Data = Pie`, `FooState` objects will generally define `Data = StatePie`.

Reply:
Fair enough for Local/Global.  State is tweaking/restricting the semantic.  Where as Pie, Ownership, MemSpace and even ThreadId are organizational (type, where, etc.) State has a semantic undertone (In particular for Params, I could see this thread indexed entity being used for "read only data" (aka definitively not a state).

Reply:
I'm open to other suggestions, but since the thread-level parallelism in CUDA is about each thread operating on independent data (which we call a state), I can't think of a situation where ThreadId isn't the only appropriate way to index into something called a State...

Reply:
"ThreadPie" :) ... but probably too specific :)